recipes/serdepp/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+from conan.tools.microsoft import msvc_runtime_flag
+import conan.tools.files
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class SerdeppConan(ConanFile):
+    name = "serdepp"
+    description = "c++ serialize and deserialize adaptor library like rust serde.rs"
+    license = "MIT"
+    topics = ("yaml", "toml", "serialization", "json", "reflection")
+    homepage = "https://github.com/injae/serdepp"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    options = {
+        # keeping the option in case upstream support dynamic linking
+        "with_nlohmann_json": [True, False],
+        "with_rapidjson": [True, False],
+        "with_fmt": [True, False],
+        "with_toml11": [True, False],
+        "with_yamlcpp": [True, False],
+    }
+    default_options = {
+        "with_nlohmann_json": True,
+        "with_rapidjson": True,
+        "with_fmt": True,
+        "with_toml11": True,
+        "with_yamlcpp": True,
+    }
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder,
+                  strip_root=True)
+
+    def package_id(self):
+        self.info.header_only()
+
```

Comment:
```suggestion

    @property
    def _compilers_minimum_version(self):
        return {
            "gcc": "7",
            "Visual Studio": "15",
            "clang": "5",
            "apple-clang": "10",
        }

    def validate(self):
        compiler = self.settings.compiler
        if compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, "17")
        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)

        if not minimum_version:
            self.output.warn(f"{self.name} requires C++17. Your compiler is unknown. Assuming it supports C++17.")
        elif tools.Version(self.settings.compiler.version) < minimum_version:
            raise ConanInvalidConfiguration(f"{self.name} requires a compiler that supports at least C++17")

```

This project requires C++17 according to the readme: https://github.com/injae/serdepp#serdepp-----